runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -5270,11 +5270,577 @@ static TR::Register *inlineFPTrg1Src3(TR::Node *node, TR::InstOpCode::Mnemonic o
+       * b.ge        loopStartLabel
+       * ands        sizeLeftReg, sizeLeftReg, #0x1f
+       * b.eq        doneLabel
+       * add         baseReg, baseReg, #8
+       * residualLabel:
+       */
+      auto align8LabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, align8Label);
+
+      TR::LabelSymbol *loopStartLabel = generateLabelSymbol(cg);
+
+      generateCompareImmInstruction(cg, node, sizeLeftReg, 32, false);
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, residualLabel, TR::CC_LT);
+      auto loopHeaderLabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, loopHeaderLabel);
+      auto preBiasInstr = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::subimmx, node, baseReg, baseReg, 8);
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::subimmw, node, sizeLeftReg, sizeLeftReg, 32);
+      auto loopStartLabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, loopStartLabel);
+      for (int32_t i = 0; i < 4; i++)
+         {
+         TR::MemoryReference *mref = TR::MemoryReference::createWithDisplacement(cg, baseReg, (i + 1) * 8);
+         /* Using pre-index load for the last one. */
+         generateTrg1MemInstruction(cg, (i == 3) ? TR::InstOpCode::ldrprex : TR::InstOpCode::ldrimmx, node, tmpReg, mref);
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::crc32cx, node, inputReg, inputReg, tmpReg);
+         }
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::subsimmw, node, sizeLeftReg, sizeLeftReg, 32);
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, loopStartLabel, TR::CC_GE);
+      /* sizeLeft &= 0x1f. immr = 0, imms = 4 for 0x1f */
+      generateLogicalImmInstruction(cg, TR::InstOpCode::andsimmw, node, sizeLeftReg, sizeLeftReg, false, 4);
+      auto branchToDoneInstr = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, doneLabel, TR::CC_EQ);
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, baseReg, baseReg, 8);
+      auto residualLabelInstr = generateLabelInstruction(cg, TR::InstOpCode::label, node, residualLabel);
+      if (debugObj)
+         {
+         debugObj->addInstructionComment(align8LabelInstr, "align8Label");
+         debugObj->addInstructionComment(loopHeaderLabelInstr, "loopHeaderLabel");
+         debugObj->addInstructionComment(loopStartLabelInstr, "loopStartLabel");
+         debugObj->addInstructionComment(preBiasInstr, "pre-bias base register by 8");
+         debugObj->addInstructionComment(branchToDoneInstr, "If sizeLeft is 0, jump to doneLabel");
+         debugObj->addInstructionComment(residualLabelInstr, "residualLabel");
+         }
+
+      /*
+       * residual 8 byte loads
+       *
+       * cmp         sizeLeftReg, #8
+       * b.lt        residual4Label
+       * tbz         sizeLeftReg, #4, left8_15
+       * tbz         sizeLeftReg, #3, left16_23
+       * left24_31:
```

Comment:
Label `left24_31` remains in the comment.

Reply:
I would like to leave the label in the comment as it is for readability.

Reply:
OK.